File: ieee_hello_world_3/python/D.py
def read():
    return int(input())
def read_list():
    return [int(i) for i in input().split()]


def solve():
    n = read()
    a = sorted([read_list() for i in range(n)], key=lambda x: x[1])
    result = 0
    l = 0
    r = n-1
    while l <= r:
        result = max(result, a[l][1]+a[r][1])
        if a[l][0] > a[r][0]: 
            a[l][0] -= a[r][0] 
            r -= 1
        elif a[l][0] < a[r][0]:
            a[r][0] -= a[l][0] 
            l += 1
        else:
            l += 1
            r -= 1
    return result
    


# for i in range(read()):   
print(solve())

# B D E G H I 

File: ieee_hello_world_3/cpp/D.cpp
#include "cstdlib"
#include "cstring"
#include "cmath"
#include "iostream"
#include "stack"
#include "vector"
#include "array"
#include "queue"
#include "algorithm"
#include "map"
#include "set"
#include "unordered_map"
#include "unordered_set"
#include "numeric"
#include "string"
#include "iomanip"
#include "climits"
#include "functional"
#include "cctype"

using namespace std; 

#define all(a) a.begin(), a.end()
using ll = long long;
#define nl '\n'
#define MOD 1000000007

#define dbg(value) cerr << "#" #value << ": " << (value) << nl;

template<typename type> 
struct matrix : vector<vector<type>> {
    matrix(int N, int M, type Val = {}) : 
        ::vector<vector<type>>(N, vector<type>(M, Val)){}     
}; 



void solve() {
    int n; cin >> n;
    vector<pair<int, int>> v(n);
    int result = 0;
    for (int i=0; i<n; i++) {
        cin >> v[i].second >> v[i].first;
    }   
    sort(all(v));
    int l = 0;
    int r = n-1;
    while (l<=r) {
        result = max(result, v[l].first+v[r].first);
        if (v[l].second == v[r].second) {
            l++;
            r--;
        } else if (v[l].second > v[r].second) {
            v[l].second -= v[r].second;
            r--;
        } else {
            v[r].second -= v[l].second;
            l++;
        }
    }
    cout << result << nl;
}

int main() {
    #ifndef ONLINE_JUDGE 
        freopen("input.txt", "r", stdin);
        // freopen("output.txt", "w", stdout);
    #else
        ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
    #endif
    // preprocess();
    int t=1;
    // cin >> t;
    for (int i=0; i<t; i++){
        solve();
    }

}





